test/unit/math/rev/mat/vectorize/apply_scalar_unary_test.cpp

```diff
@@ -0,0 +1,18 @@
+#include <stan/math/rev/core/var.hpp>
+#include <stan/math/rev/scal/fun/exp.hpp>
+#include <stan/math/rev/mat/vectorize/apply_scalar_unary.hpp>
+#include <test/unit/math/prim/mat/vectorize/foo_base_test.hpp>
+#include <test/unit/math/prim/mat/vectorize/expect_types.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_values.hpp>
+#include <test/unit/math/rev/mat/vectorize/expect_errors.hpp>
+#include <gtest/gtest.h>
+
+/**
+ * this tests that the expect_values test works on a mock function 
+ */
+TEST(MathRevMatVectorize, applyScalarUnaryMock) {
```

Comment:
Thanks for showing me how to do it yesterday. I'm trying what you suggested, but I'm getting errors because the underlying test functions are named the same and are being redefined due to the include statements. I was worried that I would run into this issue if we wanted a mix function calling all tests. 

Today, I tried adding a stan::math::var as a dummy template parameter, but that caused other issues. I could:

a) Add a dummy argument to the function call (e.g. `expect_scalar_value(stan::math::var x)`) 
b) Add a different number of dummy template parameters using int
c) Rename functions
d) Some other approach 

I'm wondering what the best would be.